string distribution
float64 proximity
float64 distance
float64 press
